src/transform/src/join_implementation.rs

```diff
@@ -557,15 +569,46 @@ mod delta_queries {
             // A viable delta query requires that, for every order,
             // there is an arrangement for every input except for
             // the starting one.
-            if !orders
+            let arrangement_counts = orders
                 .iter()
-                .all(|o| o.iter().skip(1).all(|(c, _, _)| c.arranged))
-            {
-                return Err(TransformError::Internal(String::from(
-                    "delta plan not viable",
-                )));
+                .map(|o| o.iter().skip(1).filter(|(c, _, _)| c.arranged).count())
+                .collect::<Vec<_>>();
+            let expected_number_of_arrangements = inputs.len() - 1;
+            let missing_arrangements: std::collections::BTreeSet<usize> = orders
+                .iter()
+                .flat_map(|o| {
+                    o.iter()
+                        .skip(1)
+                        .filter_map(|(c, _, input)| if c.arranged { None } else { Some(*input) })
+                })
+                .collect();
+            let insufficiently_arranged_orders = missing_arrangements.len();
+
+            if insufficiently_arranged_orders > 0 {
+                ::tracing::info!("{insufficiently_arranged_orders} orders out of {} have insufficient arrangements ({}; missing {} distinct arrangements: {})",
```

Comment:
Spelling this out because not everybody is familiar with Rust multiline strings:

```rust
::tracing::info!(
    "{insufficiently_arranged_orders} orders out of {} have insufficient arrangements \
     ({}; missing {} distinct arrangements: {})",
```

will give you equivalent output. FYI I think that if you use something like

```rust
::tracing::info!(
    insufficiently_arranged_orders = insufficiently_arranged_orders,
    total_orders = total_orders,
    keyX = valueX,
    keyY = valueY,
    "insufficient arrangements",
)
```

will produce output that is easier to analyze with a script. If the main goal is to get a sense of the current state in our production environments I think this format will be much better because we can just process the output with a JSON parser. 

You can prefix values with `%` to use a `Display` implementation or `?` to use a `Debug` implementation for values that are not supported as first-class citizens. See [the docs](https://docs.rs/tracing/latest/tracing/#recording-fields) for details.

Reply:
Done! Will be mindful of line length in the future.